code/datums/loadout/loadout_helper.dm

```diff
@@ -1,25 +1,101 @@
+/proc/is_savable_in_loadout(obj/item/saved_item, datum/loadout/loadout, obj/item/storage/master_storage)
+	//Some items are allowed to bypass everything
+	if(is_type_in_typecache(saved_item, GLOB.bypass_loadout_check_item))
+		return TRUE
+
+	//If the master storage has it for free on initialisation, we bypass the check
+	for(var/type in master_storage?.starting_items)
+		if(saved_item.type == type)
+			return TRUE
+	
 	if(ishandful(saved_item))
 		return is_handful_savable(saved_item)
+	
+	//We check if the item is in a public vendor for free
 	for(var/type in GLOB.loadout_linked_vendor)
 		for(var/datum/vending_product/item_datum AS in GLOB.vending_records[type])
 			if(item_datum.product_path == saved_item.type)
 				return TRUE
+
+	//If we can't find it for free, we then look if it's in the essential set of each job
+	var/list/job_specific_list = GLOB.loadout_role_essential_set[loadout.job]
+	if(islist(job_specific_list) && job_specific_list[saved_item.type] > loadout.unique_items_list[saved_item.type])
+		loadout.unique_items_list[saved_item.type] += 1
+		return TRUE
+
+	//If we can't find it for free, we then look if it's in a job specific vendor and if we can buy that category
+	job_specific_list = GLOB.job_specific_clothes_vendor[loadout.job]
+	var/list/item_info = job_specific_list[saved_item.type]
+	if(item_info && buy_category(item_info[1], loadout))
+		loadout.clothes_item_list[saved_item.type] += 1
+		return TRUE
+
+	//We check for specific construction stack items
+	if((loadout.job == SQUAD_LEADER || loadout.job == SQUAD_ENGINEER) && isitemstack(saved_item))
+		var/obj/item/stack/stack_saved = saved_item
+		var/base_amount = 0
+		var/base_price = 0
+		if(istype(stack_saved, /obj/item/stack/sheet/metal) && loadout.job == SQUAD_ENGINEER)
+			base_amount = 10
+			base_price = METAL_PRICE_IN_GEAR_VENDOR
+		else if(istype(stack_saved, /obj/item/stack/sheet/plasteel) && loadout.job == SQUAD_ENGINEER)
+			base_amount = 10
+			base_price = PLASTEEL_PRICE_IN_GEAR_VENDOR
+		else if(istype(stack_saved, /obj/item/stack/sandbags_empty))
+			base_amount = 25
+			base_price = SANDBAG_PRICE_IN_GEAR_VENDOR
+		if(base_amount && (round(stack_saved.amount / base_amount) * base_price <= loadout.job_points_available))
+			var/points_cost = round(stack_saved.amount / base_amount) * base_price
+			loadout.job_points_available -= points_cost
+			loadout.priced_items_list[saved_item.type] = points_cost
+			return TRUE
+	//If it was not in a job specific clothes vendor, we try to use marine points to buy it
+	var/list/listed_products = GLOB.job_specific_points_vendor[loadout.job]
+	if(!listed_products)
+		return FALSE
+	for(var/item_type in listed_products)
+		if(saved_item.type != item_type)
+			continue
+		item_info = listed_products[item_type]
+		if(loadout.job_points_available < item_info[3])
+			return FALSE
+		loadout.job_points_available -= item_info[3]
+		loadout.priced_items_list[saved_item.type] = item_info[3]
```

Comment:
explain this magic number index

Reply:
it's literraly where the price is put in vendors, i didn't do it